internal/data/data.go

```diff
@@ -0,0 +1,102 @@
+package data
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
+
+	"github.com/DanWlker/remind/internal/config"
+)
+
+// This does not create the file if it doesn't exist
+func GetTodoFromFile(filename string) ([]TodoEntity, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("os.Open: %w", err)
+	}
+	defer f.Close()
+
+	var (
+		items []TodoEntity
+		dec   = yaml.NewDecoder(f)
+	)
+
+	if err := dec.Decode(&items); err != nil {
+		return nil, fmt.Errorf("yaml.Unmarshal: %w", err)
+	}
+
+	return items, nil
+}
+
+func WriteTodoToFile(filename string, todoList []TodoEntity) error {
+	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o644)
+	if err != nil {
+		return fmt.Errorf("os.OpenFile: %w", err)
+	}
+	defer f.Close()
+
+	enc := yaml.NewEncoder(f)
```

Comment:
Likewise, stream the YAML-marshaling output directly to the file instead of into memory first.